### PR TITLE
samples/tests: Fix checking for board names to not use ${BOARD} variable

### PR DESCRIPTION
**Pull request by nordicjm**
Fixes using outdated hwmv1 board name checking with proper board
    defines

and

    samples: shields: x_nucleo_iks01a3: Fix deprecated config fragment
    
    Fixes uses an old deprecated way of setting a Kconfig fragment
    file for this sample

`samples/drivers/ipm/ipm_esp32` and `samples/subsys/ipc/rpmsg_service` have been omitted as these need to be reworked

### CODE REVIEW

File: samples/boards/nrf/nrf53_sync_rtc/CMakeLists.txt [hunk 1]
```diff
@@ -7,11 +7,11 @@ cmake_minimum_required(VERSION 3.20.0)
 
 find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE})
 
-if(("${BOARD}" STREQUAL "nrf5340dk") OR
-	("${BOARD}" STREQUAL "nrf5340bsim_nrf5340_cpuapp"))
-	message(INFO " ${BOARD} used for Application Core")
+if(CONFIG_BOARD_NRF5340DK_NRF5340_CPUAPP OR
+   CONFIG_BOARD_NRF5340BSIM_NRF5340_CPUAPP OR
```
Comment (pdgendt) [inline samples/boards/nrf/nrf53_sync_rtc/CMakeLists.txt:11]:
```suggestion
   CONFIG_BOARD_NRF5340BSIM_NRF5340_CPUAPP)
```

File: samples/boards/nrf/nrf53_sync_rtc/net/CMakeLists.txt [hunk 2]
```diff
@@ -7,11 +7,11 @@ cmake_minimum_required(VERSION 3.20.0)
 
 find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE})
 
-if(("${BOARD}" STREQUAL "nrf5340dk") OR
-	("${BOARD}" STREQUAL "nrf5340bsim_nrf5340_cpunet"))
-	message(INFO " ${BOARD} used for Network Core")
+if(CONFIG_BOARD_NRF5340DK_NRF5340_CPUNET OR
+   CONFIG_BOARD_NRF5340BSIM_NRF5340_CPUNET OR
```
Comment (pdgendt) [inline samples/boards/nrf/nrf53_sync_rtc/net/CMakeLists.txt:11]:
```suggestion
   CONFIG_BOARD_NRF5340BSIM_NRF5340_CPUNET)
```

Comment (nordicjm) [reply samples/boards/nrf/nrf53_sync_rtc/net/CMakeLists.txt]: Fixed

File: samples/boards/nrf/nrf53_sync_rtc/CMakeLists.txt (hunk 1 above)
Comment (nordicjm) [reply samples/boards/nrf/nrf53_sync_rtc/CMakeLists.txt]: Fixed

**Review by erwango — DISMISSED** [review]
Approved for STM32 parts

**Review by PavelVPV — DISMISSED** [review]
mesh sample change approved

**Review by pdgendt — DISMISSED** [review]

**Review by erwango — APPROVED** [review]
Approved for shield related part, thanks

**Review by pdgendt — APPROVED** [review]

**Review by tejlmand — APPROVED** [review]

**Pull request merged**